R/object_name_linters.R

```diff
@@ -29,14 +29,15 @@ object_name_linter <- function(styles = c("snake_case", "symbols")) {
     xml <- source_file$full_xml_parsed_content
 
     xpath <- paste0(
-      # Left hand assignments
-      "//expr[SYMBOL or STR_CONST][following-sibling::LEFT_ASSIGN or following-sibling::EQ_ASSIGN]/*",
-
-      # Or
-      " | ",
-
-      # Right hand assignments
-      "//expr[SYMBOL or STR_CONST][preceding-sibling::RIGHT_ASSIGN]/*",
+      # assignments
+      "//SYMBOL[",
+      " not(preceding-sibling::OP-DOLLAR)",
+      " and ancestor::expr[",
```

Comment:
I have a feeling `ancestor` may be too aggressive without more conditions. Running the comparison script now to try and confirm.

Reply:
Sounds good. I tested "//parent::expr" before, which produced false positives (detecting symbols on the other side of assigments).
"ancestor" was the only axis that passed all the tests.